common/authentication/postgresql/metadata.go

```diff
@@ -87,6 +102,117 @@ func (m *PostgresAuthMetadata) InitWithMetadata(meta map[string]string, opts Ini
 	return nil
 }
 
+// buildConnectionString builds the connection string from the metadata.
+// It supports both DSN-style and URL-style connection strings.
+// Metadata fields override existing values in the connection string.
+func (m *PostgresAuthMetadata) buildConnectionString() (string, error) {
+	if m.ConnectionString == "" {
+		return "", errors.New("connection string is required")
+	}
+
+	metadata := m.getConnectionStringMetadata()
+
+	if strings.HasPrefix(m.ConnectionString, "postgres://") || strings.HasPrefix(m.ConnectionString, "postgresql://") {
+		return m.buildURLConnectionString(metadata)
+	}
+	return m.buildDSNConnectionString(metadata)
+}
+
+func (m *PostgresAuthMetadata) buildDSNConnectionString(metadata map[string]string) (string, error) {
+	connectionString := ""
```

Comment:
Is it designed intentionally not to allow for - examples sake, a connection string with a host and/or password to override?

Reply:
wdym? the goal is to start building from connection string if it exists, override matching params passed in metadata and then append all the remaining

Reply:
The changelog at the time was confusing, this looks fine to me now!